tools/please_pex/pex/pex_main.py

```diff
@@ -156,7 +156,10 @@ def read_text(self, filename):
                 return zf.read(name).decode(encoding="utf-8")
 
     def locate_file(self, path):
-        raise RuntimeError("This distribution has no real file system")
+        return zipfile.Path(
+            self._pex_file,
+            at=os.path.join(self._prefix, path) if self._prefix else path,
+        )
```

Comment:
importlib.metadata's `files` method is _slow_ - it calls `locate_file` for every file listed in the distribution's `RECORD` file, then confirms that each of those paths exists by calling `exists` on the `zipfile.Path` that is returned. This triggers a sequential scan for the file's local file header in the pex file for every file in the distribution.

`//test:distribution_metadata_test`, which calls `files` for pygments, takes 0.9 seconds to run on Python >= 3.12. In earlier Python versions, importlib.metadata takes it as given that the files listed in `RECORD` actually exist. I guess this is a cheap operation for real files on a real file system and they weren't thinking of any use case more exotic than that. I'm not sure there's much we can or should do about this here.